src/test/groovy/org/prebid/server/functional/tests/SetUidSpec.groovy

```diff
@@ -31,4 +54,177 @@ class SetUidSpec extends BaseSpec {
     }
+
+    def "PBS setuid should remove expired uids cookie"() {
+        given: "Default SetuidRequest"
+        def request = SetuidRequest.defaultSetuidRequest
+        def uidsCookie = UidsCookie.defaultUidsCookie.tap {
+            def uidWithExpiry = defaultUidWithExpiry.tap {
+                expires = ZonedDateTime.now(Clock.systemUTC()).minusDays(2)
+            }
+            tempUIDs = [(GENERIC): uidWithExpiry]
+        }
+
+        when: "PBS processes setuid request"
+        def response = prebidServerService.sendSetUidRequest(request, uidsCookie)
+
+        then: "Response shouldn't contain uids cookie"
+        assert response.uidsCookie.tempUIDs.size() == 0
+    }
+
+    def "PBS setuid should populate uids cookie with non priority bidder"() {
+        given: "Setuid request"
+        def request = SetuidRequest.defaultSetuidRequest.tap {
+            uid = UUID.randomUUID().toString()
+        }
+        def uidsCookie = UidsCookie.defaultUidsCookie.tap {
+            tempUIDs = [(APPNEXUS): defaultUidWithExpiry,
+                        (RUBICON) : defaultUidWithExpiry]
+        }
+
+        when: "PBS processes setuid request"
+        def response = prebidServerService.sendSetUidRequest(request, uidsCookie)
+
+        then: "Response should contain none priority uids"
+        assert response.uidsCookie.tempUIDs.size() == 2
+    }
+
+    def "PBS setuid should return prioritized uids bidder when size is full"() {
+        given: "PBS config"
+        def bidderGeneric = GENERIC
+        PrebidServerService prebidServerService = pbsServiceFactory.getService(PBS_CONFIG +
+                ["cookie-sync.pri"                  : bidderGeneric.value])
+
+        and: "Setuid request"
+        def request = SetuidRequest.defaultSetuidRequest.tap {
+            it.bidder = bidderGeneric
+            uid = UUID.randomUUID().toString()
+        }
+        def uidsCookie = UidsCookie.defaultUidsCookie.tap {
+            tempUIDs = [(APPNEXUS): defaultUidWithExpiry,
+                        (RUBICON) : defaultUidWithExpiry]
+        }
+
+        when: "PBS processes setuid request"
+        def response = prebidServerService.sendSetUidRequest(request, uidsCookie)
+
+        then: "Response should contain uids cookies"
+        assert response.uidsCookie.tempUIDs[GENERIC]
+    }
+
+    def "PBS setuid should remove earliest expiration bidder when size is full"() {
+        given: "PBS config"
+        PrebidServerService prebidServerService = pbsServiceFactory.getService(PBS_CONFIG +
+                ["cookie-sync.pri"                  : GENERIC.value])
+
+        and: "Setuid request"
+        def request = SetuidRequest.defaultSetuidRequest.tap {
+            uid = UUID.randomUUID().toString()
+        }
+        def uidsCookie = UidsCookie.defaultUidsCookie.tap {
+            def uidWithExpiry = defaultUidWithExpiry.tap {
+                expires = ZonedDateTime.now(Clock.systemUTC()).plusDays(2)
```

Comment:
This should be oldest uids, right? Please differentiate both uids somehow. You're relying on clock to be really granular.